### PR TITLE
Creation of poi and uploading image done

### PR DESCRIPTION
**Pull request by 0figueiredo**
Creation of poi done and uploading image also.

### CODE REVIEW

File: queries/createNewPoi.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
-const createNewPoi = 'INSERT INTO point_of_interest SET ?, picture_id=1';
+const createNewPoi = `INSERT INTO point_of_interest SET
+?,
+creation_date = NOW(), 
+author_id = (SELECT id FROM user WHERE name=?), 
+picture_id= (SELECT id from picture WHERE id=?);`;
```
Comment (erabin05) [inline queries/createNewPoi.js]: ```picture_id=id```do exactly the same thing

File: queries/createNewPoi.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
-const createNewPoi = 'INSERT INTO point_of_interest SET ?, picture_id=1';
+const createNewPoi = `INSERT INTO point_of_interest SET
+?,
+creation_date = NOW(), 
+author_id = (SELECT id FROM user WHERE name=?),
```
Comment (erabin05) [inline queries/createNewPoi.js]: couldn't you directly get the auth id rather than find it with its name ?

File: routes/pointofinterest/route.js [hunk 3]
```diff
@@ -41,17 +46,74 @@ router.use((req, res, next) => {
...
+      res.status(500).send(`Erreur lors de la création du point d'intéret : ${err}`);
     } else {
-      res.sendStatus(200);
+      // Get the id of the poi created previously
+      const resultId = { poi_id: result.insertId };
+      connection.query(linkNewlyCreatedPoiWithKeyword, [keyword, resultId], (error) => {
+        if (error) {
+          res.status(500).send(`Erreur lors de l'ajout du thème : ${error}`);
+        } else {
+          connection.query(insertGradesNewPoi, [grades, authorName, resultId.poi_id], (e) => {
```
Comment (erabin05) [inline routes/pointofinterest/route.js]: not ```e```for ```error``` error_grades would be better for exemple (and if you change this one like this, change the other)

File: routes/pointofinterest/route.js [hunk 4]
```diff
@@ -41,17 +46,74 @@ router.use((req, res, next) => {
...
   });
 });
 
+
+// upload picture
+router.post('/picture', (req, res) => {
+  let formData = new formidable.IncomingForm();
+  formData.parse(req, function (err, fields, files) {
+    let olpath = files.file.path;
+    let newpath = `./public/images/${files.file.name}`;
```
Comment (erabin05) [inline routes/pointofinterest/route.js]: if it doesn't change use ```const```

**Review by erabin05 — CHANGES_REQUESTED** [review]
- ```SELECT id from table WHERE id=?``` will always give you the same result as your input id
- To insert foreign like ```author_id``` in ```createNewPoi``` you could send directly the id from the front rather than send the name and have to find the id with SQL
- e is not a right name for an error
- use ```const```instead of ```let```when you can

File: queries/createNewPoi.js (hunk 1 above)
Comment (0figueiredo) [reply queries/createNewPoi.js]: correction done

File: queries/createNewPoi.js (hunk 2 above)
Comment (0figueiredo) [reply queries/createNewPoi.js]: set this way to use the username later on when authentication will be implemented.

File: routes/pointofinterest/route.js (hunk 3 above)
Comment (0figueiredo) [reply routes/pointofinterest/route.js]: changed but in camelCase to follow linter instructions

File: routes/pointofinterest/route.js (hunk 4 above)
Comment (0figueiredo) [reply routes/pointofinterest/route.js]: changed

**Review by erabin05 — APPROVED** [review]
Ok

**Pull request merged**